queries.sql

```diff
@@ -7,4 +7,41 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
 SELECT name,escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = TRUE;
 SELECT * FROM animals WHERE NOT name='Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+/*2ND PULL REQUEST */
+
+BEGIN TRANSACTION;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+
+UPDATE animals SET SPECIES = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+BEGIN TRANSACTION;
+DELETE FROM animals;
+ROLLBACK;
+BEGIN TRANSACTION;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT SP1;
+UPDATE animals SET weight_kg= weight_kg * -1;
+ROLLBACK TO SP1;
+UPDATE animals SET weight_kg= weight_kg * -1 WHERE weight_kg < 0;
+COMMIT;
+
+
+/* ANSWER QUESTIONS */
+
+/*HOW MANY ANIMALS ARE THERE?*/
+SELECT COUNT(*) FROM animals;
+
+/*HOW MANY ANIMALS HAVE NEVER TRIED TO ESCAPE*/
+SELECT COUNT(*) FROM animals WHERE escape_attempts=0;
+/* WHAT IS THE AVG WEIGHT OF THE ANIMALS */
+SELECT AVG(weight_kg) FROM animals;
+/*WHO ESCAPES THE MOST, NEUTURED OR NOT NEUTERED ANIMALS*/
+SELECT neutered, SUM(escape_attempts) FROM animals GROUP BY neutered HAVING SUM(escape_attempts) >= ALL(SELECT SUM(escape_attempts) FROM animals GROUP BY neutered);
```

Comment:
- [ ] I think here since the question is Who escapes the most, neutered or not neutered animals? so you want the most it means the max so you can use som thing like this:

>  SELECT neutered, MAX(escape_attempts) FROM animals
  GROUP BY neutered;